runtime/tr.source/trj9/z/codegen/J9S390PrivateLinkage.cpp

```diff
@@ -2282,7 +2282,7 @@ TR::S390PrivateLinkage::buildVirtualDispatch(TR::Node * callNode, TR::RegisterDe
             generateRXInstruction(cg(), TR::InstOpCode::getLoadOpCode(), callNode, RegRA, generateS390MemoryReference(classReg, offset, cg()));
          cursor = generateRIInstruction(cg(), TR::InstOpCode::getLoadHalfWordImmOpCode(), callNode, RegZero, offset);
          }
-      else if (offset > -524288)
+      else
```

Comment:
For cases where offset is <= -524288, are we relying on long displacement support to generate the correct memory reference offset?  Have we tested that it works for this case?

Reply:
Yes that is precisely what we're relying on. The long displacement logic will effectively generate what we have in the (now deleted) `else` path. This is just code duplication. I've tested on internal builds that this does indeed function as expected.